### ISSUE TITLE
org.assertj.core.data.MapEntry is not compatible with java Map.Entry

### ISSUE THREAD

**Issue by Blahord** [joel-costigliola/assertj-core#606]
I noticed, that the hashCode method of org.assertj.core.data.MapEntry is not implemented correctly.
As the documentation of Map.Entry (https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#hashCode--) says:

> Returns the hash code value for this map entry. The hash code of a map entry e is defined to be:
>```
(e.getKey()==null   ? 0 : e.getKey().hashCode()) ^
     (e.getValue()==null ? 0 : e.getValue().hashCode())
```

The MapEntry hashCode looks this way:
```
int result = 1;
result = HASH_CODE_PRIME * result + hashCodeFor(key); //HASH_CODE_PRIME = 31
result = HASH_CODE_PRIME * result + hashCodeFor(value);
return result;
```
So
```
Collections.singletonMap(1, 10).entrySet().iterator().next().equals(Assertions.entry(1, 10))
```
gives `true`, but
```
Collections.singletonMap(1, 10).entrySet().iterator().next().hashCode() == Assertions.entry(1, 10).hashCode()
```

give `false`.
As the documentation says
`e1.equals(e2)` imples `e1.hashCode() == e2.hashCode()`, which is not true, for the current MapEntry implementation.

I tried to make a pull request, but your build-server don't want to build it. Don't know what i did wrong. :(

**Comment by joel-costigliola** [comment]
I agree, good catch, I have also found that `MapEntry` was not symmetric in regard to java `Map.Entry`

**Comment by joel-costigliola** [comment]
Fixed by this commit 3118bf043099fba418bc908388afc0fbd25cf772, it would be great if you could confirm that.

Thanks for reporting this.
